.circleci/defaults.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+# shared variables between the different services
+#
+# CircleCI set values:
+#   $SERVER - points to the millennium build server
+#   $AWS_SERVER - points to the aws manager instance
+#   $CI_DIR - home directory on build server
+#   $CI_AWS_DIR - home directory on aws
+
+##################
+# SHARED FUNCTIONS
+##################
+
+# shared across build server and aws
+
+# copy files from server to server
+# $1 - src
+# $2 - dest
+copy () {
+    rsync -avzp -e 'ssh' --exclude '.git' $1 $2
+}
+
+# run command over ssh
+# $1 - server login (user@IP)
+# $2 - command
+run_impl () {
+    ssh -o "StrictHostKeyChecking no" -t $1 $2
+}
+
+# run script over ssh
+# $1 - server login (user@IP)
+# $2 - local script
+# $2 - arguments to the script
+run_script_impl () {
+    ssh -o "StrictHostKeyChecking no" -t $1 'bash -s -l' < $2 "$3"
+}
+
+# build server calls
+
+# run command on build server
+# $1 - command
+run () {
+    run_impl $SERVER "$@"
+}
+
+# run script on build server
+# $1 - script
+# $1 - arguments to the script
+run_script () {
+    run_script_impl $SERVER $1 $2
+}
+
+# remove the work dir on the build server
+clean () {
+    run "rm -rf $REMOTE_WORK_DIR"
+}
+
+# aws calls
+
+# run command on aws server
+# $1 - command
+run_aws () {
+    run_impl $AWS_SERVER "$@"
+}
+
+# run script on aws server
+# $1 - script
+# $1 - arguments to the script
+run_script_aws () {
+    run_script_impl $AWS_SERVER $1 $2
+}
+
+# run script on aws server but detach after running
+# $1 - session name
+# $2 - local script path
+# $3 - script name
+run_detach_script_aws () {
+    # remove old script if it exists
+    run_impl $AWS_SERVER "rm -rf $REMOTE_AWS_WORK_DIR/$3"
+    # copy new script to run to work dir
+    copy $2/$3 $AWS_SERVER:$REMOTE_AWS_WORK_DIR/
+    # run script and detach
+    # NOTE: for some reason without re-invoking screen after this command it breaks
+    # NOTE: AND, screen -list returns false by default (even if there is a session running)
+    run_impl $AWS_SERVER "screen -S CI-$1-SESSION -dm $REMOTE_AWS_WORK_DIR/$3; screen -list || true"
+}
+
+# remove the work dir on the aws server
+clean_aws () {
+    run_aws "rm -rf $REMOTE_AWS_WORK_DIR"
+}
+
```

Comment:
This AWS stuff isn't use here, right?
Its up to you if you want to keep it. I don't foresee us ever needing to use this feature in testchipip

Reply:
I'll remove it. I forgot to delete this.